include/chemfiles/files/MemoryFile.hpp

```diff
@@ -0,0 +1,92 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#ifndef CHEMFILES_MEMORY_FILES_HPP
+#define CHEMFILES_MEMORY_FILES_HPP
+
+#include "chemfiles/File.hpp"
+#include "chemfiles/files/MemoryBuffer.hpp"
+
+#include <istream> // Needed for clang
+#include <ostream> // Needed for clang
+#include <iosfwd>
+
+namespace chemfiles {
+
+/// Simple TextFile implementation, that wraps memory for reading only.
+class MemoryFileReader final: public TextFileImpl {
+    class vector_buffer_reader : public std::basic_streambuf<char, std::char_traits<char>> {
```

Comment:
The use of `iostream` is not needed, but I like the fact that it is a subclass of `TextFileImpl` instead of `TextFile`. This allows the memory files to be created by the `TextFile` constructor.